js/views/conversation_view.js

```diff
@@ -241,6 +278,10 @@
             this.model.messageCollection.add(message, {merge: true});
             message.setToExpire();
 
+            if (this.lastSeenIndicator) {
+                this.lastSeenIndicator.increment(1);
+            }
+
```

Comment:
Yes, keychange notices will trigger this event, which might mess up the count. Maybe better to always set the number directly from this.model.get('unreadCount') rather than incrementing?